handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -1927,6 +1925,12 @@ final void initHandshakeException(Throwable cause) {
         return FINISHED;
     }
 
+    private SSLEngineResult.HandshakeStatus mayFinishHandshake(
+            SSLEngineResult.HandshakeStatus hs, int bytesConsumed, int bytesProduced) throws SSLException {
+        return hs == NEED_UNWRAP && bytesProduced > 0 || hs == NEED_WRAP && bytesConsumed > 0 ?
```

Comment:
@Scottmitch can you just add some comment explaining what is going here so that it is easy to understand when reading the code in a few months ?

Reply:
let me do this a followup so we can merge this quickly